sklearn/ensemble/tests/test_voting.py

```diff
@@ -674,3 +680,97 @@ def test_get_features_names_out_classifier_error():
     )
     with pytest.raises(ValueError, match=msg):
         voting.get_feature_names_out()
+
+
+# Metadata Routing Tests
+# ======================
+
+
+@pytest.mark.parametrize("Estimator", [VotingClassifier, VotingRegressor])
+def test_routing_passed_metadata_not_supported(Estimator):
+    """Test that the right error message is raised when metadata is passed while
+    not supported when `enable_metadata_routing=False`."""
+
+    X = np.array([[0, 1], [2, 2], [4, 6]])
+    y = [1, 2, 3]
+
+    if is_classifier(Estimator):
+        SubEst = ConsumingClassifier
+    else:
+        SubEst = ConsumingRegressor
+
+    with pytest.raises(
+        ValueError, match="is only supported if enable_metadata_routing=True"
+    ):
+        Estimator(["clf", SubEst()]).fit(X, y, sample_weight=[1, 1, 1], metadata="a")
+
+
+@pytest.mark.usefixtures("enable_slep006")
+@pytest.mark.parametrize("Estimator", [VotingClassifier, VotingRegressor])
+@pytest.mark.parametrize("prop", ["sample_weight", "metadata"])
```

Comment:
With introducing `prop` I have tried to parametrize this test, while still keeping `sample_weight` as a param name, because it might get some special treatment in some estimators (not sure if this is the case in  `ConsumingClassifier` and `ConsumingRegressor`, too).